src/pyvesync/vesyncbulb.py

```diff
@@ -653,121 +958,133 @@ def set_color_mode(self, color_mode: str) -> bool:
+
+        Returns
+        -------
+        bool
+            True if call was successful, False otherwise
+        """
+        arg_list = ['brightness', 'color_temp', 'color_saturation',
+                    'color_hue', 'color_mode', 'color_value']
+        toggle_on = True
+        for val in arg_list:
+            if locals()[val] is not None:
+                toggle_on = False
+        if toggle_on:
+            self.turn_on()
+
+        # If any HSV color values are passed,
+        # set HSV status & ignore other values
+        # Set Color if hue, saturation or value is set
+        if color_hue is not None or color_value is not None or \
+                color_saturation is not None:
+            return self.set_hsv(color_hue, color_saturation, color_value)
+
         # initiate variables
-        color_mode_api = ''
-        should_force = 1
-        if (color_temp is None and
-                color_saturation is None and
-                color_hue is None and
-                color_mode is None and
-                color_value is None):
-
-            if brightness is not None:
-                should_force = 0
-            else:
-                # Turn on without any extra parameters
-                if self.toggle('on'):
-                    self.device_status = 'on'
-                    return True
-                self.device_status = 'off'
-                self.connection_status = 'offline'
-                logger.debug('%s offline', self.device_name)
-                return False
+        request_dict = {
+            "force": 1,
+            "colorMode": '',
+            "brightness": '',
+            "colorTemp": '',
+            "hue": "",
+            "saturation": "",
+            "value": ""
+        }
 
-        # """Set brightness of multicolor bulb."""
-        brightness_api: Union[str, int] = ''
+        force_list = ['colorTemp', 'saturation', 'hue', 'colorMode', 'value']
         if brightness is not None:
-            if not self.dimmable_feature:
-                logger.debug('%s is not dimmable', self.device_name)
-                return False
-            if not isinstance(brightness, int):
-                logger.error(
-                    'Error: brightness value should be a integer '
-                    'number between 1 and 100 OR None')
-                return False
-            if brightness < 1 or brightness > 100:
-                logger.warning(
-                    'Warning: brightness value should be '
-                    'between 1 and 100')
-            # ensure brightness is between 0 and 100
-            brightness = max(1, (min(100, brightness)))
-            brightness_api = int(brightness)
-
-        # """Set White Temperature of Bulb in pct (1 - 100)."""
-        color_temp_api: Union[str, int] = ''
+            brightness_update = self._validate_brightness(brightness)
+
+            if not all(locals().get(k) is None for k in force_list):
```

Comment:
Is this line double negation logic, maybe getting unexpected result?

```suggestion
            if all(locals().get(k) is None for k in force_list):
```

Reply:
I just tested my change suggestion. it worked. 👍

Reply:
Great, thank you for testing so quickly. 😀 I will push changes. My goal is to implement tests that test the consistency of calls to assist in refactoring but I've been tight on time so the patchwork will have to do for now.

Reply:
I wish I knew how to propperly implement new tests when I first added the valceno bulbs, but mocking an api was beyond me, I still have a lot learn. And coding is not my main job.